scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2631,6 +2632,13 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
               targs.fold(ref)(x => autoEndPos(sid)(Term.ApplyType(ref, x))),
               autoPos(argumentPatterns().reduceWith(Pat.ArgClause.apply))
             )
+          } else if (at[Equals] && isNamedTupleOk) {
+            next()
+            sidToken match {
+              case ident: Ident => Pat.Assign(Term.Name(ident.value), noSeqWithNamed.pattern())
+              case _ => syntaxError(s"pattern must be an identifier: $sid", at = currToken)
+
+            }
```

Comment:
instead of new code above, we can put it below:

```
sid match {
  case name: Term.Name if isNamedTupleOk && acceptOpt[Equals] =>
    Pat.Assign(name, noSeqWithNamed.pattern())
  // rest is as before
  case name: Term.Name.Quasi => ...
}
```

that way, we handle possible targs, sid being Quasi or Select, etc.

Reply:
sorry, forgot to "Save" this comment earlier.

Reply:
Good idea, changed